app/components/UI/ApproveTransactionReview/index.js

```diff
@@ -1040,13 +1044,14 @@ class ApproveTransactionReview extends PureComponent {
       frequentRpcList,
       providerRpcTarget,
     } = this.props;
-    const { showBlockExplorerModal, address } = this.state;
+    const { showBlockExplorerModal, address, learnMoreURL } = this.state;
 
     const styles = this.getStyles();
     const closeModal = () => {
-      showVerifyContractDetails();
+      !learnMoreURL && showVerifyContractDetails();
       this.setState({
         showBlockExplorerModal: !showBlockExplorerModal,
+        learnMoreURL: undefined,
```

Comment:
[nit] I think null would be preferred here since we are setting the value 
```suggestion
        learnMoreURL: null,
```

Reply:
Done. https://github.com/MetaMask/metamask-mobile/pull/6619/commits/082a6f9ade63b0530ac308b728db7074fcd167e7